File: email-service/src/main/java/com/digitalbooks/model/Book.java
package com.digitalbooks.model;

import java.time.LocalDate;

import org.springframework.format.annotation.DateTimeFormat;

import com.digitalbooks.common.BookCategory;
import com.fasterxml.jackson.annotation.JsonFormat;

import lombok.Data;

@Data
public class Book {
	private Integer id;
	private Author author;
	private String tittle;
	private BookCategory category;
	private String image;
	private Double price;
	@DateTimeFormat(pattern = "dd/MM/yyyy")
	@JsonFormat(pattern = "dd/MM/yyyy")
	private LocalDate publishDate;
	private String publisher;
	private Boolean active;
	private String content;
}

File: frontend/src/entity/book.ts
export interface Book {
  id:number;
  tittle: string;
  category:String,
  price: number;
  publishDate: any;
  active: Boolean;
  content: String;
  purchased:boolean;
  pdf:any;
  image:any;
  author:any;
}
export interface SearchBook {
  tittle: string;
  category:string,
  price: number;
  publisher:string;
  authorName: string;
}
